include/galsim/SBFourierSqrtImpl.h

```diff
@@ -0,0 +1,80 @@
+/* -*- c++ -*-
+ * Copyright (c) 2012-2015 by the GalSim developers team on GitHub
+ * https://github.com/GalSim-developers
+ *
+ * This file is part of GalSim: The modular galaxy image simulation toolkit.
+ * https://github.com/GalSim-developers/GalSim
+ *
+ * GalSim is free software: redistribution and use in source and binary forms,
+ * with or without modification, are permitted provided that the following
+ * conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions, and the disclaimer given in the accompanying LICENSE
+ *    file.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions, and the disclaimer given in the documentation
+ *    and/or other materials provided with the distribution.
+ */
+
+#ifndef GalSim_SBFourierSqrtImpl_H
+#define GalSim_SBFourierSqrtImpl_H
+
+#include "SBProfileImpl.h"
+#include "SBFourierSqrt.h"
+
+namespace galsim {
+
+    class SBFourierSqrt::SBFourierSqrtImpl : public SBProfile::SBProfileImpl
+    {
+    public:
+        SBFourierSqrtImpl(const SBProfile& adaptee, const GSParamsPtr& gsparams);
+        ~SBFourierSqrtImpl() {}
+
+        // xValue() not implemented for SBFourierSqrt.
+        double xValue(const Position<double>& p) const;
+
+        SBProfile getObj() const { return _adaptee; }
+
+        std::complex<double> kValue(const Position<double>& k) const;
+
+        double maxK() const { return _adaptee.maxK(); }
+        double stepK() const { return _adaptee.stepK(); }
```

Comment:
I think technically, this should be `_adaptee.stepK() / sqrt(2.)`.  Perhaps you could add a unit test that checks the (at least approximate) equivalence of these values for `galsim.AutoConvolve(galsim.FourierSqrt(obj))` and `obj`.  

They should be equivalent conceptual objects, so a few tests that they draw similar results would probably also be appropriate.